trieste/acquisition/optimizer.py

```diff
@@ -35,23 +35,33 @@
 AcquisitionOptimizer = Callable[[SP, AcquisitionFunction], TensorType]
 """
 Type alias for a function that returns the single point that maximizes an acquisition function over
-a search space. For a search space with points of shape S, and acquisition function with input shape
-[...] + S output shape [..., 1], the :data:`AcquisitionOptimizer` return shape should be [1] + S.
+a search space. For a search space with points of shape [D], and acquisition function with input
+shape [..., B, D] output shape [..., 1], the :data:`AcquisitionOptimizer` return shape should be
+[B, D].
 """
 
 
 @singledispatch
-def optimize(space: Box | DiscreteSearchSpace, target_func: AcquisitionFunction) -> TensorType:
+def optimize(
+    space: Box | DiscreteSearchSpace, target_func: AcquisitionFunction, num_query_points: int = 1
+) -> TensorType:
     """
     :param space: The space of points over which to search, for points with shape [D].
-    :param target_func: The function to maximise, with input shape [..., D] and output shape
-        [..., 1].
-    :return: The point in ``space`` that maximises ``target_func``, with shape [1, D].
+    :param target_func: The :const:`AcquisitionFunction` to maximise, with input shape [..., B, D]
+            and output shape [..., 1].
+    :param num_query_points: The number of points to acquire.
+    :return: The points in ``space`` that together maximize ``target_func``, with shape [B, D].
     """
 
 
 @optimize.register
-def _discrete_space(space: DiscreteSearchSpace, target_func: AcquisitionFunction) -> TensorType:
+def optimize_discrete(
+    space: DiscreteSearchSpace, target_func: AcquisitionFunction, num_query_points: int = 1
+) -> TensorType:
+
+    if num_query_points > 1:
+        raise ValueError("Batch optimisation not yet supported for discrete search spaces")
```

Comment:
actually, it should! if I'm not mistaken we can apply the same operation to discrete spaces: `space ** num_query_points`